unison-core/src/Unison/Name.hs

```diff
@@ -134,9 +144,33 @@ suffixes (Name' names) =
   map (Name' . NonEmpty.fromList) (NonEmpty.init (NonEmpty.tails names))
 
 makeAbsolute :: Name -> Name
-makeAbsolute n | toText n == "."                = Name ".."
-               | Text.isPrefixOf "." (toText n) = n
-               | otherwise                      = Name ("." <> toText n)
+makeAbsolute n | toText n == "." = Name ".."
+               | isAbsolute n    = n
+               | otherwise       = Name ("." <> toText n)
```

Comment:
This could just be setting the flag, though I actually wonder how it's being used, and whether it should be used at all.

Reply:
It does seem to be used, maybe not a lot, but I couldn't easily refactor the call sites without thinking :)